### PR TITLE
mgr/dashboard_v2: add  sparkline and monitor  components

### PR DESCRIPTION
**Pull request by tspmelo**
This PR includes both the sparkline and the monitor components.

You can see both components in the image below, 
the sparkline is being reused in each row of the table:
192 168 1 136_4200_ fhd 1

New version:
192 168 1 136_4200_ fhd 2

`Signed-off-by: Tiago Melo <tmelo@suse.com>`

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/tests/test_monitor.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class MonitorTest(ControllerTestCase):
+    @classmethod
+    def setup_test(cls):
+        Monitor._cp_config['tools.authenticate.on'] = False
+
+    def test_ping(self):
+        self.getPage("/api/monitor")
+        self.assertStatus('200 OK')
+        # self.assertJsonBody({'echo': 'Hello World'})
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard_v2/tests/test_monitor.py]: Remove this line

**Comment by jecluis** [comment]
So, if this PR requires the thing you introduced in #57, and #57 is not actually making use of said code, why not submitting those commits alongside this PR?

**Comment by tspmelo** [comment]
merged #57 into this PR.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor/monitor.component.html [hunk 2]
```diff
@@ -0,0 +1,115 @@
+<!-- Page Header -->
+<h1 class="name">
+  Monitors
+</h1>
+
+<!-- Main content -->
+<div class="row">
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor/monitor.component.html]: Should be replace with the data table component.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  ];
+
+  public labels: Array<any> = [];
+
+  constructor() {}
+
+  ngOnInit() {}
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.datasets[0].data = changes['data'].currentValue;
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:72]: Is "currentValue" an array?

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    this.datasets = [...this.datasets];
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:73]: Is this needed to get an update from Angular?

**Review by Devp00l — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts (hunk 3 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:72]: In this case, yes.
It represents the current value of the input `data`

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts (hunk 4 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:73]: It seems like it, if you don't use the `...` the plugins will not update.

Comment (jecluis) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:73]: @tspmelo mind elaborating a bit? :)

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor/monitor.component.html (hunk 2 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor/monitor.component.html]: I'm now using the cdTable on both quorum tables.

**Comment by ricardoasmarques** [comment]
Shouldn't we add a breadcrumb like we have on other pages (e.g. Hosts / Pools)?

File: src/pybind/mgr/dashboard_v2/tests/test_monitor.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/tests/test_monitor.py]: What about actually testing the body? (and rename test_ping)

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor-sparkline-data.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'monitorSparklineData'
+})
+export class MonitorSparklineDataPipe implements PipeTransform {
+  transform(value: any, args?: any): any {
+    // let result = '';
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor-sparkline-data.pipe.ts]: Commented code should be removed.

**Comment by tspmelo** [comment]
Just pushed a new version.

Changes:
- Applied suggested changes
- Add 2 new table templates, sparkline and router link.
- Add new performance counter page, so we can have a dedicated page to show that information
- Move datatable module to another module, to fix a circular dependency error

**Review by Devp00l — DISMISSED** [review]

**Review by sebastian-philipp — DISMISSED** [review]

**Comment by votdev** [comment]
When the browser window is small, the paging input field in the datatable tool header shouldn't be resized. There should be displayed the number '10'.
bildschirmfoto vom 2018-02-15 14-15-55

 ```
--- src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss	(date 1518694121000)
+++ src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss	(date 1518700914000)
@@ -76,6 +76,7 @@
   }
   .input-group.dataTables_paginate {
     width: 8%;
+    min-width: 75px;
   }
 }
 ```

**Review by LenzGr — DISMISSED** [review]

**Review by votdev — APPROVED** [review]

**Pull request merged**